xcm/pallet-xcm/src/migration.rs

```diff
@@ -36,8 +37,11 @@ pub mod v1 {
 		}
 
 		fn on_runtime_upgrade() -> Weight {
+			CurrentMigration::<T>::put(VersionMigrationStage::default());
```

Comment:
Does this make sense to run even if the upgrade is a noop (`StorageVersion::get::<Pallet<T>>() != 0`)?

Reply:
I tried not to alter the existing logic here, which I think is fine.
If you run the migration with try-runtime from v1 it will fail because of the assert in the pre-upgrade, indicating that the fix is to remove this migration from the migrations passed to `frame_executive::Executive`